server/region.go

```diff
@@ -59,6 +60,50 @@ func (r *regionInfo) String() string {
 	return fmt.Sprintf("%+v", *r)
 }
 
+func (r *regionInfo) DiffPeer(other *regionInfo) string {
```

Comment:
I think for DiffPeers, you should return two []Peer for added and removed, not string here.

Btw, seem that it is complex here, so I prefer logging like `peers changed %v -> %v` with Peers directly.